### DISCUSSION TITLE
Vercel not working with cloudflare proxy (Custom Domains)

### DISCUSSION THREAD

**Discussion by amirisongithub** [vercel/community#5317] [Help] [answered]
### Summary

I use cloudflare as my dns provider and when I tried using vercel with it, it just told me “configuration invalid”, I know that cloudflare proxy is causing the issue because as soon as I turned off the cloudflare proxy switch it said that it’s valid, I have looked on the cloudflare and vercel integration doc and I’ve excluded any vercel stuff from cloudflare rules/firewall but the issue is still not fixed

Sorry if this isn’t lots of info because that’s literally all that is.

(I have seen A TON of research and none of the other discussion matched/solved my issue)

### Steps to Reproduce

-Use cloudflare as your dns provider or whatever it’s called
-Enable the proxy tick on the vercel dns

**Comment by amirisongithub** [comment]
Just commenting to make this to the top since in 20 hours nobody responded

**Comment by alexllao** [comment]
![imagen](https://github.com/vercel/community/assets/8004573/e81e09ec-8fe1-47ed-b255-6328d18b6d6f)

**Comment by ryoid** [comment, marked as answer]
Ensure that your **SSL mode** is **Full**, mine was Flexible and did not work.

https://vercel.com/docs/integrations/cloudflare#using-cloudflare-as-a-reverse-proxy
